proto/options.proto

```diff
@@ -190,4 +190,5 @@ message Options {
   optional uint32 height_precision = 43 [default = 0];                    // Number of digits precision for heights returned
   optional bool roundabout_exits = 44 [default = true];                   // Whether to announce roundabout exit maneuvers
   optional bool linear_references = 45;                                   // Include linear references for graph edges returned in certain responses.
+  optional bool match_on_restrictions = 46;                               // User input parameter to allow map-match on restricted edges
```

Comment:
i know it says dont review yet but i wanted to mention something design related. it may be that we dont actually need an option to allow what you want but instead rather use the correct costing which doesnt care about restrictions. for example, you could specify a costing type of "none" and completely allow map matching to any part of the network.

there is one downside to that approach which is that the timing of the maneuvers on the route will be strange and potentially the maneuver might not make sense. so if that is important then that is not the way to go.

the other consideration we need to think about and i think the most important one is that costing determines whether or not restrictions matter. if we want to disable restrictions we dont want to do it just for map matching we want to do it anywhere a costing is used. for that reason, it should be a field under CostingOptions and we should have dynamic cost parse this similar to how it already parses the speed types. it can then short circuit the restriction check when its called

Reply:
Forgot to note this earlier. @yuzhe and I discussed this feedback this morning and are moving this into `CostingOptions`, updating the option parser, etc.